packages/path_provider/path_provider_foundation/test/path_provider_foundation_test.dart

```diff
@@ -119,5 +119,27 @@ void main() {
       expect(
           pathProvider.getExternalStoragePaths(), throwsA(isUnsupportedError));
     });
+
+    test('getContainerPath', () async {
+      const String appGroupIdentifier = 'group.example.test';
+      if (Platform.isIOS) {
```

Comment:
@stuartmorgan is there a suggested way to mock out the platform in these tests?  I know how to do it in the flutter_tool but not here.

Reply:
It usually doesn't come up here given the ways plugins are structured; I would just write a trivial object-based wrapper around the `Platform` check and make it injectable in the constructor. (And then move the `PathProviderFoundation` construction out of `setUp`, which is something I general prefer anyway.)

Reply:
Can you explain what that means? 🙈  Constructor for what, the `PathProviderFoundation`?

Reply:
Yes, exactly; it would look like [this](https://github.com/flutter/plugins/blob/15d799b893679336fe42b59b0c4b3ca5f5d40ec4/packages/url_launcher/url_launcher_windows/lib/url_launcher_windows.dart#L15-L16), with the parameter being some little class you define (e.g., `PathProviderPlatformProvider`) that is itself `@visibleForTesting`. The `Platform` call would move into that helper class, so you could replace it with a `FakePathProviderPlatformProvider` in tests that gives whatever answer the test case requires.